packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -0,0 +1,352 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'colors.dart';
+import 'theme.dart';
+
+// Fractional offset from 1/4 screen below the top to fully on screen.
+final Tween<Offset> _kBottomUpTween = Tween<Offset>(
+  begin: const Offset(0.0, 0.25),
+  end: Offset.zero,
+);
+
+// Slides the page upwards and fades it in, starting from 1/4 screen
+// below the top.
+class _GenericPageTransition extends StatelessWidget {
+  _GenericPageTransition({
+    Key key,
+    @required Animation<double> routeAnimation,
+    @required this.child,
+  }) : _positionAnimation = _kBottomUpTween.animate(CurvedAnimation(
+         parent: routeAnimation, // The route's linear 0.0 - 1.0 animation.
+         curve: Curves.fastOutSlowIn,
+       )),
+       _opacityAnimation = CurvedAnimation(
+         parent: routeAnimation,
+         curve: Curves.easeIn,
+       ),
+       super(key: key);
+
+  final Animation<Offset> _positionAnimation;
+  final Animation<double> _opacityAnimation;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    // TODO(ianh): tell the transform to be un-transformed for hit testing
+    return SlideTransition(
+      position: _positionAnimation,
```

Comment:
can you pass `transformHitTests: false` to this widget and remove the TODO while you're here?

Reply:
Done.

Reply:
I undid this change. It cause a few tests to fail and in the interest of limiting the scope of this PR, I've put the TODO back for now.